debian/control

```diff
@@ -4,7 +4,8 @@ Section: misc
 Priority: optional
 Build-Depends: config-package-dev,
                debhelper-compat (= 12),
-               golang
+               golang,
+               nodejs (>= 16)
```

Comment:
we need to support Ubuntu 18.04LTS (bionic), the version used at the time seems to be v8 https://packages.ubuntu.com/bionic/nodejs.

Reply:
As a general comment, assuming that nodejs is only used for development as the web app will still be served through out the go binaries, wondering whether we can ship the  "angular application build" as well with the debian package rather than build it when installing the package. This way, only the developers need to install nodejs and other relevant tools.

In such case, the challenge is to make sure that for every commit the "build" is always updated, for which we could add a Github action that verifies it.

Reply:
Is the nodejs version that important? It's not actually being used for anything other than development and to manage the angular and other packages (nvm). So it should be safe to request a version greater than or equal to the one available on Ubuntu 18.04.

I'd prefer to keep building the angular app as part of the package build process instead of adding build artifacts to the repository.

Reply:
> Is the nodejs version that important?
Yes, angular-cli which is a build tool for angular app requires active LTS or maintenance LTS version of Node.js. I could lower to to 14(which is the lowest lts version). But it should use at least version 14. (https://angular.io/guide/setup-local#prerequisites)
But it could be installed by adding ppa(like dockerfile in the change or workflows file), is it unacceptable? BTW, even in the latest ubuntu LTS, only 12.x version of nodejs is avilable. (https://packages.ubuntu.com/jammy/nodejs)

I'd prefer to build the angular app as a part of full build as well..

Reply:
Using a ppa pushes the burden of the custom nodejs version to users instead, which is not ideal.

Because this is a build-time dependency and not a run-time dependency, we might be able to get away with downloading and using a custom version of nodejs in the debian package builder script. The [nodejs website](https://nodejs.org/en/download/) provides a binary download link which looks pretty self-contained with `node` and `npm` executables plus shared libraries. The user doesn't need to deal with this later since nodejs isn't used for the constructed debian package.

This seems comparable to the way we are already downloading some go packages dynamically in our setup. Adding a checksum comparison to the nodejs release should validate we're downloading what we expect.